lib/versioning/semver-cargo/index.js

```diff
@@ -0,0 +1,39 @@
+const semver = require('../semver');
+
+function convertToCaret(item) {
+  if (semver.isVersion(item)) {
+    return '^' + item.trim();
+  }
+  return item.trim();
+}
+
+// FIXME: Partial versions like '1.2' don't get converted to '^1.2'
```

Comment:
Is `1.2` a *version* (i.e. same as `1.2.0`) or is it a version (i.e. same as `1.2.x`)?

Reply:
As per https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements:

```
^1.2 := >=1.2.0 <2.0.0
^0.2 := >= 0.2.0 < 0.3.0
```

Reply:
In npm partial versions are equivalent to wildcard versions, see [npm reference](https://www.npmjs.com/package/semver#x-ranges-12x-1x-12-):

> A partial version range is treated as an X-Range, so the special character is in fact optional.
>
>    "" (empty string) := * := >=0.0.0
>    1 := 1.x.x := >=1.0.0 <2.0.0
>    1.2 := 1.2.x := >=1.2.0 <1.3.0

[cargo reference](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html) doesn't mention partial versions, but it is a reasonable assumption that cargo also treats partial versions as wildcard versions and version requirements like `"0.2.*"` and `"0.2"` are locked to the same version when I run `cargo update`.